nostr-client-plus/src/bin/publisher.rs

```diff
@@ -144,3 +154,42 @@ async fn run() -> Result<()> {
     println!("Done: {} jobs sent", jobs_sent);
     Ok(())
 }
+
+// Return a new event based on entry, when entry is None it's a
+// PoW event, otherwise it's Classification event
+fn construct_event(entry: Option<RawDataEntry>, sender: Sender) -> UnsignedEvent {
```

Comment:
I believe the most simple thing to do, was to go through all the events in `entries` first (that would have also solved the `clone` problem above), and then the remaining jobs treated as PoW, so this function would not be dealing with both, but maybe a function for building classification event and another one for PoW.
But this is your code, so feel free to do it as you please.

However, for reasons like "readability" and "efficiency" (not wasting CPU time), just make at the very beginning the code branch: if its PoW do this, if it's classification, do that.
Why do we want to call random 10 thousand times (for example), and then discarding the result because we are actually creating 10 thousand classification jobs.

Why should we complicate things like the `println` I commented below, where by mistake we print an id that is wrong?